Scouting/src/main/java/ca/team2706/scouting/mcmergemanager/powerup2018/StatsEngine.java

```diff
@@ -786,8 +846,39 @@ private void fillInTeleopStats(TeamStatsReport teamStatsReport) {
                         teamStatsReport.numMatchesNoDeadness++;
                     }
                 }
+            }
+
+            // Finish cycles
+            cubeCycle.endTime = 0;
+            if(inSwitchCycle) {
+                teamStatsReport.switchAvgCycleTime += cubeCycle.getCycleTime();
+                teamStatsReport.switchCycleTimes.add(cubeCycle.getCycleTime());
+                cyclesInThisMatch.cycles.add(cubeCycle.clone(Cycle.CycleType.SWITCH));
 
-                // TODO: i don't think i need to finish cycles
+                if(match.preGameObject.matchNumber > LATER_MATCH_START) {
+                    teamStatsReport.switchLaterCycles++;
+                    teamStatsReport.switchAvgLaterCycleTime += cubeCycle.getCycleTime();
+                }
+            }
+            if(inScaleCycle) {
+                teamStatsReport.scaleAvgCycleTime += cubeCycle.getCycleTime();
+                teamStatsReport.scaleCycleTimes.add(cubeCycle.getCycleTime());
+                cyclesInThisMatch.cycles.add(cubeCycle.clone(Cycle.CycleType.SCALE));
+
+                if(match.preGameObject.matchNumber > LATER_MATCH_START) {
+                    teamStatsReport.scaleLaterCycles++;
+                    teamStatsReport.scaleAvgLaterCycleTime += cubeCycle.getCycleTime();
+                }
+            }
```

Comment:
Awesome, this looks almost right, but [last year this bit of code](https://github.com/FRC2706/MCMergeManager/blob/88523f324232d5c78fae88398a5257d37c12ec4e/Scouting/src/main/java/ca/team2706/scouting/mcmergemanager/steamworks2017/StatsEngine.java#L734) looked like:

```
if (inGearCycle) {
    currGearCycle.endTime = 135;
    currGearCycle.success = false;
    cyclesInThisMatch.cycles.add(currGearCycle);
}
```

Reply:
I thought Justin had the counter counting down towards zero, I'll double check though before merging

Reply:
Ah, so the default value of 0 is already correct?